### PR TITLE
v3.0.4(schema): Update empty security array constraint

### PR DESCRIPTION
**Pull request by jeremyfiel**
based on discussions

*  https://github.com/OAI/OpenAPI-Specification/issues/3938#issuecomment-2308874884

* https://github.com/OAI/OpenAPI-Specification/pull/4007

### CODE REVIEW

**Comment by handrews** [comment]
@jeremyfiel could you please re-target this to main?  We don't change the schemas on the branches (except by accident)

**Comment by jeremyfiel** [comment]
Oof.  Changing requires a new force push.  I'll do it later today when I get a chance

**Comment by ralfhandl** [comment]
@jeremyfiel Thanks for providing this!

Assuming this PR is to be merged _after_ publication of 3.0.4 I've converted it to draft.

File: schemas/v3.0/schema.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ properties:
       $ref: '#/definitions/Server'
   security:
     type: array
+    minItems: 1
```
Comment (ralfhandl) [inline schemas/v3.0/schema.yaml:23]: This contradicts what we say now in the spec

**Review by ralfhandl — CHANGES_REQUESTED** [review]
Why does this PR contain files except the schema.yaml?

Is it supposed to be merged after publication of 3.0.4?

Convertin

**Comment by handrews** [comment]
Closing for the same reasons that @lornajane articulated on the 3.1.1 version of this (although we look forward to incorporating this sort of idea in a different form in the future!):

> There are some problems with this pull request, but having discussed the proposed change (the newest commit), this isn't a useful change to make to our existing validation schemas since it is valid to have an empty security array. Please keep an eye on the project for more discussion about linting schemas in addition to our validation schemas.

**Pull request closed without merging**